### PR TITLE
Update test suite to use ssh-token and ssh-portal instead of ssh

### LINKED ISSUE #3065: SSH Portal support in drush alias
https://github.com/uselagoon/lagoon-charts/pull/382 adds remote cluster-local SSH endpoints.

It would be cool if these could be somehow integrated into the `drush` aliases generated by Lagoon. Ideally there would be some fallback available to the existing SSH service.

### PR DESCRIPTION
**Pull request by smlx**
# Checklist

- [ ] Affected Issues have been mentioned in the Closing issues section
- [ ] Documentation has been written/updated
- [x] PR title is ready for inclusion in changelog

This PR makes the following changes:
* switch the charts-treeish to [ssh-token](https://github.com/uselagoon/lagoon-charts/pull/514), which enables the `ssh-token` and `ssh-portal` services in CI. This change can be reverted once https://github.com/uselagoon/lagoon-charts/pull/514 is merged.
* switch the Lagoon test suite to use `ssh-token` and `ssh-portal` by default instead of `ssh`.
* update the drush configurations in the test suite to support `ssh-token`/`ssh-portal`.
* add a test suite for the existing `ssh` service called `ssh-legacy` which tests both token generation and remote shell access.

# Closing issues

Drush alias support for ssh-portal was really fixed in https://github.com/uselagoon/lagoon-service-images/pull/12, but this PR adds tests for that functionality, so effectively this closes #3065.

### CODE REVIEW

File: tests/files/drupal9-mariadb-single/.lagoon.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 docker-compose-yaml: docker-compose.yml
 
-ssh: lagoon-core-ssh.lagoon.svc:2020
+ssh: lagoon-remote-ssh-portal.lagoon.svc.cluster.local:2222
```
Comment (tobybellwood) [inline tests/files/drupal9-mariadb-single/.lagoon.yml]: THis should be the ssh token service in core.

Comment (smlx) [reply tests/files/drupal9-mariadb-single/.lagoon.yml]: Does drush only use this field to get a token?

File: tests/files/drupal9-mariadb-single/assets/lagoon.site.yml [hunk 2]
```diff
@@ -1,8 +1,8 @@
 '*':
-  host: lagoon-core-ssh.lagoon.svc
+  host: lagoon-core-ssh-token.lagoon.svc
```
Comment (shreddedbacon) [inline tests/files/drupal9-mariadb-single/assets/lagoon.site.yml]: this needs to be the `lagoon-remote-ssh-portal`

File: tests/files/drupal9-mariadb-single/.lagoon.yml (hunk 1 above)
Comment (tobybellwood) [reply tests/files/drupal9-mariadb-single/.lagoon.yml]: yes, that's my understanding

File: tests/files/drupal9-mariadb-single/assets/lagoon.site.yml (hunk 2 above)
Comment (smlx) [reply tests/files/drupal9-mariadb-single/assets/lagoon.site.yml]: ok trying again

**Comment by smlx** [comment]
This is now ready for review. It should be merged after https://github.com/uselagoon/lagoon-charts/pull/514

**Review by tobybellwood — APPROVED** [review]
I'm happy with this, it covers the old and the new nicely. I've merged the supporting charts PR into core, so the TREEISH ref can be updated when the merge conflict is fixed.

**Comment by smlx** [comment]
This PR incorporates https://github.com/uselagoon/lagoon/pull/3358 because otherwise the corrupt attribute value causes ssh-portal to fail.

IMO the other PR should be reviewed/merged before this one, then I can remove that commit from this PR.

**Comment by tobybellwood** [comment]
good point!

**Comment by smlx** [comment]
Okay, rebased now that #3358 is merged, once it goes green should be good to go.

**Pull request merged**